lib/home_page.dart

```diff
@@ -8,15 +24,275 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
+  //needs to be initialized in case the subscription gets something new before load is done
+  List<Model> _items = <Model>[];
+  bool _fullyLoaded = false;
+
+  @override
+  void initState() {
+    load();
+    final int profileId = SupabaseManager.getCurrentProfile()!.id!;
+    SupabaseManager.supabaseClient.channel('public:messages').on(
+      RealtimeListenTypes.postgresChanges,
+      ChannelFilter(event: 'INSERT', schema: 'public', table: 'messages', filter: 'sender_id=neq.$profileId'),
+      (payload, [ref]) {
+        if (payload['new']['sender_id'] != profileId) {
+          _loadNewMessage(payload['new']['id']);
+        }
+      },
+    ).on(
+      RealtimeListenTypes.postgresChanges,
+      ChannelFilter(event: 'UPDATE', schema: 'public', table: 'messages', filter: 'sender_id=neq.$profileId'),
+      (payload, [ref]) {
+        if (payload['new']['read'] == true) {
+          setState(() {
+            _items =
+                _items.where((Model element) => element is RideEvent || element.id != payload['new']['id']).toList();
```

Comment:
I would prefer `element is! Message`, that should be better for the future, in case we add other types of info